topydo/ui/CommandLineWidget.py

```diff
@@ -30,12 +35,36 @@ def _blur(self):
 
     def _emit_command(self):
         urwid.emit_signal(self, 'execute_command', self.edit_text)
+        self._save_to_history()
         self.clear()
 
+    def _save_to_history(self):
+        if len(self.edit_text) > 0:
+            if self.history is None:
+                self.history = []
+            self.history.append(self.edit_text)
+        self.history_pos = len(self.history)
+        self.history_tmp = self.history[:]
+        self.history_tmp.append('')
+
+    def _history_move(self, p_step):
+        if self.history is not None:
+            self.history_tmp[self.history_pos] = self.edit_text
+            self.history_pos = (self.history_pos + p_step) % len(self.history_tmp)
```

Comment:
I don't think that `self.history_pos + p_step` should ever evaluate to a negative value, that would happen when pressing down while not having moved up. You'd suddenly see the oldest entry in history, which unexpected (at least for me). Is this the behavior in prompt mode (or bash/zsh)?